File: Python/Task23/n13579.py
# https://inf.reshuege.ru/problem?id=13579

def R(n, k):
	if n == k:
		return 1
	elif n > k:
		return 0
	else:
		return R(n + 1, k) + R(n + 2, k) + R(n + 3, k)

r1 = R(1, 8);
r2 = R(8, 15);
print(r1 * r2)

File: C++/Task23/n13579/n13579.cpp
// https://inf.reshuege.ru/problem?id=13579

#include <iostream>

int R(int n, int k)
{
	if (n == k)
	{
		return 1;
	}
	else if (n > k)
	{
		return 0;
	}
	else
	{
		return R(n + 1, k) + R(n + 2, k) + R(n + 3, k);
	}
}

int main()
{
	int r1 = R(1, 8);
	int r2 = R(8, 15);
	std::cout << r1 * r2 << std::endl;
	return 0;
}